readthedocs/projects/tasks.py

```diff
@@ -122,10 +131,20 @@ def pull_cached_environment(self):
             _, tmp_filename = tempfile.mkstemp(suffix='.tar')
             remote_fd = storage.open(filename, mode='rb')
             with open(tmp_filename, mode='wb') as local_fd:
-                local_fd.write(remote_fd.read())
-
-            with tarfile.open(tmp_filename) as tar:
-                tar.extractall(self.version.project.doc_path)
+                for chunk in remote_fd.chunks():
+                    local_fd.write(chunk)
+
+            safe_makedirs(self.version.project.doc_path)
+            # Using ``tar`` command because ``tarfile`` has a memory leak when
+            # extracting big files (2.5Gb)
```

Comment:
I'm not 100% convinced that this is true. I isolated the problem but I wasn't able to find that `.extractall` consumes more memory than `tar` command when it's _outside_ read the docs environment yet.

Reply:
This was not the problem in the end. The problem was IO and copying the big file multiple times.